packages/story-editor/src/components/library/panes/shopping/shoppingPane.js

```diff
@@ -13,81 +13,204 @@
+  searchTerm: PropTypes.string,
+};
 
 function ShoppingPane(props) {
-  const isEnabled = useFeature('shoppingIntegration');
-  const [product, setProduct] = useState(null);
+  const isShoppingIntegrationEnabled = useFeature('shoppingIntegration');
+  const speak = useLiveRegion('assertive');
+  const [searchTerm, setSearchTerm] = useState('');
+  const [isMenuFocused, setIsMenuFocused] = useState(false);
+  const [products, setProducts] = useState([]);
+  const {
+    actions: { getProducts },
+  } = useAPI();
+
+  const { currentPageProducts } = useStory(({ state: { currentPage } }) => {
+    return {
+      currentPageProducts: currentPage?.elements
+        ?.filter(({ type }) => type === 'product')
+        .map(({ id, product }) => ({
+          elementId: id,
+          productId: product?.productId,
+        })),
+    };
+  });
+
+  const getProductsByQuery = useCallback(
+    async (value = '') => {
+      try {
+        setProducts(await getProducts(value));
+      } catch (err) {
+        setProducts([]);
+      }
+    },
+    [getProducts]
+  );
+
+  const debouncedProductsQuery = useDebouncedCallback((value) => {
+    getProductsByQuery(value);
+  }, 300);
+
+  const onSearch = useCallback(
+    (evt) => {
+      const value = evt.target.value;
+      if (value !== searchTerm) {
+        setSearchTerm(value);
+        debouncedProductsQuery(value);
+      }
+    },
+    [searchTerm, setSearchTerm, debouncedProductsQuery]
+  );
+
+  useEffect(() => {
+    getProductsByQuery();
+  }, [getProductsByQuery]);
+
+  const handleInputKeyPress = useCallback((event) => {
+    const { key } = event;
+    if (key === 'ArrowDown' || key === 'Tab') {
+      setIsMenuFocused(true);
+    }
+  }, []);
+
+  const handleFocus = () => {
+    setIsMenuFocused(false);
+  };
+
+  const { deleteElementById } = useStory(({ actions }) => ({
+    deleteElementById: actions.deleteElementById,
+  }));
 
   const { insertElement } = useLibrary((state) => ({
     insertElement: state.actions.insertElement,
   }));
 
-  const insertProduct = useCallback(() => {
-    insertElement('product', {
-      width: 25,
-      height: 25,
-      product,
-    });
-    setProduct(null);
-  }, [insertElement, product]);
+  const insertProduct = useCallback(
+    (product) => {
+      insertElement('product', {
+        width: 25,
+        height: 25,
+        product,
+      });
+
+      const PRODUCT_ADDED_TEXT = sprintf(
+        /* translators: %s: product title added to page.*/
+        __('%s added to page', 'web-stories'),
+        product.productTitle
+      );
+
+      speak(PRODUCT_ADDED_TEXT);
+    },
+    [insertElement, speak]
+  );
+
+  const deleteProduct = useCallback(
+    (product) => {
+      const element = currentPageProducts.find(
+        (item) => item.productId === product.productId
+      );
+      if (element) {
+        deleteElementById({ elementId: element.elementId });
+      }
+    },
+    [deleteElementById, currentPageProducts]
+  );
```

Comment:
What if you have the same product multiple times on a single page? Say you tagged a photo, someone with earrings and tagged each ear.

Reply:
AFAIK you can only add product once on a page.

Reply:
That said ... your right --- I can copy paste a dot 🤔  -- will need to figure out a way to handle.